tensorflow/compiler/mlir/tosa/transforms/legalize_utils.cc

```diff
@@ -1007,5 +1007,36 @@ tosa::MulOp CreateMulOpAndInfer(PatternRewriter& rewriter, Operation* op, Type r
       input2, getConstTensor<int8_t>(rewriter, op, shift, {1}).value());
 }
 
+bool checkUniqueConstantScatterIndices(ShapedType indices_type,
```

Comment:
I suppose "negative tests" are not a thing in TF, i.e. there wouldn't be a way to test that the logic in this function produces correct behaviour in the case of overlapping indices?

Reply:
added testcase